src/datasets/load.py

```diff
@@ -1836,28 +1837,23 @@ def dataset_module_factory(
                     token=download_config.token,
                     timeout=100.0,
                 )
-            except Exception as e:  # noqa catch any exception of hf_hub and consider that the dataset doesn't exist
-                if isinstance(
-                    e,
-                    (
-                        OfflineModeIsEnabled,
-                        requests.exceptions.ConnectTimeout,
-                        requests.exceptions.ConnectionError,
-                    ),
-                ):
-                    raise ConnectionError(f"Couldn't reach '{path}' on the Hub ({type(e).__name__})")
-                elif "404" in str(e):
-                    msg = f"Dataset '{path}' doesn't exist on the Hub or cannot be accessed"
-                    raise DatasetNotFoundError(msg + f" at revision '{revision}'" if revision else msg)
-                elif "401" in str(e):
-                    msg = f"Dataset '{path}' doesn't exist on the Hub or cannot be accessed"
-                    msg = msg + f" at revision '{revision}'" if revision else msg
-                    raise DatasetNotFoundError(
-                        msg
-                        + f". If the dataset is private or gated, make sure to log in with `huggingface-cli login` or visit the dataset page at https://huggingface.co/datasets/{path} to ask for access."
-                    )
-                else:
-                    raise e
+            except (
+                OfflineModeIsEnabled,
+                requests.exceptions.ConnectTimeout,
+                requests.exceptions.ConnectionError,
+            ) as e:
+                raise ConnectionError(f"Couldn't reach '{path}' on the Hub ({e.__class__.__name__})") from e
+            except GatedRepoError as e:
+                raise DatasetNotFoundError(
+                    f"Dataset '{path}' is a gated dataset on the Hub. Visit the dataset page at https://huggingface.co/datasets/{path} to ask for access."
```

Comment:
maybe mention that the user may have to login ?

Reply:
addressed in https://github.com/huggingface/datasets/pull/6959/commits/62350f5cddfe056c71ca2bb760250f76b5a49394

Reply:
Thanks.